lib/improver/nbhood/circular_kernel.py

```diff
@@ -0,0 +1,361 @@
+                axes.append(cube.coord_dims(coord_name)[0])
+        except CoordinateNotFoundError:
+            raise ValueError("Invalid grid: projection_x/y coords required")
+        for axis_index, axis in enumerate(axes):
+            fullranges[axis] = ranges[axis_index]
+        kernel = Utilities.circular_kernel(fullranges, ranges,
+                                           self.weighted_mode)
+        # Smooth the data by applying the kernel.
+        cube.data = scipy.ndimage.filters.correlate(
+            data, kernel, mode='nearest') / np.sum(kernel)
+        return cube
+
+    def run(self, cube, radius):
+        """
+        Call the methods required to calculate and apply a circular
+        neighbourhood.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube containing to array to apply CircularNeighbourhood processing
+            to.
+        radius : Float
+            Radius in metres for use in specifying the number of
+            grid cells used to create a circular neighbourhood.
+
+        Returns
+        -------
+        cube : Iris.cube.Cube
+            Cube containing the smoothed field after the kernel has been
+            applied.
+        """
+        ranges = convert_distance_into_number_of_grid_cells(
+            cube, radius, MAX_RADIUS_IN_GRID_CELLS)
+        cube = self.apply_circular_kernel(cube, ranges)
+        return cube
+
+
+class CircularPercentiles(object):
+    """
+    Methods for use in calculating percentiles from a 2D circular
+    neighbourhood.
+
+    A maximum kernel radius of 500 grid cells is imposed in order to
+    avoid computational ineffiency and possible memory errors.
+    """
+    def __init__(self,
+                 percentiles=PercentileConverter.DEFAULT_PERCENTILES):
+        """
+
+        Parameters
+        ----------
+
+        percentiles : list (optional)
+            Percentile values at which to calculate; if not provided uses
+            DEFAULT_PERCENTILES from percentile module.
+        """
+        self.percentiles = percentiles
+
+    def __repr__(self):
+        """Represent the configured class instance as a string."""
+        result = ('<CircularPercentiles: percentiles: {}>')
+        return result.format(self.percentiles)
+
+    def run(self, cube, ranges):
+        """
+        Method to apply a circular kernel to the data within the input cube in
+        order to derive percentiles over the kernel.
+
+        Parameters
+        ----------
+        cube : Iris.cube.Cube
+            Cube containing array to apply processing to.
+        ranges : Int
+            Number of grid cells in the x and y direction used to create
+            the kernel.
+
+        Returns
+        -------
+        outcube : Iris.cube.Cube
+            Cube containing the percentile fields.
+            Has percentile as an added dimension.
+
+        """
+        # Take data array and identify X and Y axes indices
+        try:
```

Comment:
I think this is related to the fact that the neighbourhood processing is focussed on an equal areas grid, rather than a latitude/longitude grid. However @fionaRust , as you have pointed out, `projection_x_coordinate` and `projection_y_coordinate` do not necessarily represent an equal areas grid, therefore, this test is probably invalid, and we should instead be checking that we have an equal areas grid before proceeding.;